Controller/HelperController.php

```diff
@@ -291,6 +288,28 @@ public function setObjectFieldValueAction(Request $request)
             $value = filter_var($value, FILTER_VALIDATE_BOOLEAN);
         }
 
+        // Handle entity choice association type, transforming the value into entity
+        if ('' !== $value && $fieldDescription->getType() == 'choice' && $fieldDescription->getOption('class')) {
+            // Get existing associations for current object
+            $associations = $admin->getModelManager()
+                ->getEntityManager($admin->getClass())->getClassMetadata($admin->getClass())
+                ->getAssociationNames();
+
+            if (!in_array($field, $associations)) {
+                return new JsonResponse(sprintf('Unknown association "%s", association does not exist in entity "%s", available associations are "%s".',
+                        $field, $this->admin->getClass(), implode(', ', $associations)), 404);
+            }
+
+            $value = $admin->getConfigurationPool()->getContainer()->get('doctrine')->getManager()
+                ->getRepository($fieldDescription->getOption('class'))
+                ->find($value);
+
+            if (!$value) {
+                return new JsonResponse(sprintf('Edit failed, object with id "%s" not found in association "%s".',
+                        $originalValue, $field), 404);
```

Comment:
This is not PSR-compliant, although not detected yet. Please put each `sprintf` argument on its own line.